util/pkg/vfs/s3fs.go

```diff
@@ -124,7 +124,10 @@ func (p *S3Path) WriteFile(data io.ReadSeeker, aclObj ACL) error {
 	request.Body = data
 	request.Bucket = aws.String(p.bucket)
 	request.Key = aws.String(p.key)
-	request.ServerSideEncryption = aws.String(sse)
+	// only support SSE if a custom endpoint is not provided
+	if aws.StringValue(client.Config.Endpoint) == "" {
```

Comment:
This was breaking for my setup and not sure if others face similar issues when using a custom S3 endpoint or if anyone even enables SSE on a custom endpoint. I'm happy to remove it for this PR if you think it'll introduce issues in the upcoming release.

Reply:
Can we split it into a separate PR?  Though I believe this is only used by other clouds I don't want to turn off encryption, even though the encryption is not terribly useful given AWS holds the keys and the data.

Maybe we could configure the SSE option in the s3Context?  I'd imagine that eventually we'll recognize a `do://` VFS endpoint, even if we speak the S3 protocol to it - for the purposes of bucket discovery etc.  I thought we had a PR in the works to change the SSE option, though I can't find it right now...

Reply:
Good point I'll open a separate PR, and yes I will be adding a `do://` VFS soon!